### PR TITLE
Move schema files to centralized catalog

### PR DESCRIPTION
**Pull request by Swiddis**
### Description
Per side-channel discussion, we realized that it's going to block integration development if we need to keep our schemas in sync with each other everywhere. This PR centralizes the catalog files so that variations in schemas can be better tracked. This is a stand-in solution until the [catalog API](https://github.com/opensearch-project/opensearch-catalog/issues/28) is available at `opensearch-catalog`.

**Following this PR**, the following process for updating catalog files is advised:
1. Create a new copy of the catalog file with an updated version number in accordance with SemVer.
2. Please sync updates to catalog files back with `opensearch-catalog`.
3. Optional: inform authors of integrations using old catalog files that a new one is available. We'd like to release with all integrations using the same catalog files, and prune any redundant copies as we go. We'll need to do one final pass-through before release.

### Issues Resolved
N/A

### Check List
- [X] New functionality includes testing.
  - [X] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/adaptors/integrations/__data__/catalog/logs_vpc-1.0.0.mapping.json
Comment (YANG-DB) [inline server/adaptors/integrations/__data__/catalog/logs_vpc-1.0.0.mapping.json:1]: this should be removed - all log based integrations should use the single log mapping

File: server/adaptors/integrations/__data__/catalog/aws_s3-1.0.0.mapping.json
Comment (YANG-DB) [inline server/adaptors/integrations/__data__/catalog/aws_s3-1.0.0.mapping.json:1]: all aws mapping should reside under catalog/aws folder

**Review by YANG-DB — CHANGES_REQUESTED** [review]

File: server/adaptors/integrations/__data__/catalog/logs_vpc-1.0.0.mapping.json
Comment (Swiddis) [reply server/adaptors/integrations/__data__/catalog/logs_vpc-1.0.0.mapping.json:1]: We discussed that we need to work with these duplicates for now to account for multiple component configs. Let's work on fixing it later, as a hacky solution I can try setting the `composed_of` list dynamically, but I want to keep this PR small

File: server/adaptors/integrations/__data__/catalog/observability/logs/logs_vpc-1.0.0.mapping.json [hunk 1]
```diff
@@ -220,7 +220,7 @@
     }
   },
   "composed_of": [
-    "aws_vpc_flow",
+    "vpc_flow",
```
Comment (YANG-DB) [inline server/adaptors/integrations/__data__/catalog/observability/logs/logs_vpc-1.0.0.mapping.json:223]: all aws specific component templates should have the `aws_` prefix

File: server/adaptors/integrations/__data__/repository/aws_vpc_flow/aws_vpc_flow-1.0.0.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "displayName": "AWS VPC Flow",
   "description": "AWS VPC Flow log collector",
   "license": "Apache-2.0",
-  "type": "logs",
+  "type": "logs_vpc",
```
Comment (YANG-DB) [inline server/adaptors/integrations/__data__/repository/aws_vpc_flow/aws_vpc_flow-1.0.0.json:7]: all the `logs` index based integrations should use the same index template names `logs`
the only dynamic things that should be integration specific are:
 - index alias
 - `composed_of` component list

File: server/adaptors/integrations/__data__/repository/nginx/nginx-1.0.0.json [hunk 3]
```diff
@@ -22,14 +21,17 @@
     },
     "components": [
         {
+            "catalog_dir": "observability/logs",
```
Comment (YANG-DB) [inline server/adaptors/integrations/__data__/repository/nginx/nginx-1.0.0.json:24]: this should be configured on a higher level configuration and all should use that implicit as default location

**Review by YANG-DB — CHANGES_REQUESTED** [review]

File: server/adaptors/integrations/__data__/repository/aws_vpc_flow/aws_vpc_flow-1.0.0.json (hunk 2 above)
Comment (Swiddis) [reply server/adaptors/integrations/__data__/repository/aws_vpc_flow/aws_vpc_flow-1.0.0.json:7]: > We discussed that we need to work with these duplicates for now to account for multiple component configs. Let's work on fixing it later, as a hacky solution I can try setting the composed_of list dynamically, but I want to keep this PR small

**Comment by Swiddis** [comment]
Todo: Update front-end display to trim leading directories from categories.

Actual: If you open the VPC integration you'll see the categories listed as `aws/vpc_flow, cloud, communication, logs_vpc, aws/s3`.

Expected: Trim leading directories for `vpc_flow, cloud, communication, logs_vpc, s3`, and possibly capitalize everything + remove underscores to make it look nicer? Still a little awkward but easy to implement and a step in the right direction. `Vpc Flow, Cloud, Communication, Logs Vpc, S3`.

**Comment by Swiddis** [comment]
Status update: Postponing to after we're done with 2.9 release. This is a hefty refactor and will need some time to get done right. Will leave it as a draft for now, we can revisit it in a month or so.

**Comment by YANG-DB** [comment]
resolved in https://github.com/opensearch-project/opensearch-catalog/issues/69

**Pull request closed without merging**